frame/examples/default-config/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! <!-- markdown-link-check-disable -->
+//! # Default Config Pallet Example
+//!
+//! A simple example of a FRAME pallet that utilizes [`frame_support::derive_impl`] to
+//! implement a `DefaultConfig` for testing purposes.
+//!
+//! See the source code for `tests.rs` to see the relevant
+//! [`derive_impl`](`frame_support::derive_impl`) example.
+//!
+//! Note that this pallet makes use of `dev_mode` for ease of use, since the point of this
+//! example is the tests, not the pallet itself. See `pallet-dev-mode` for a more detailed
+//! example of the capabilities of `dev_mode`.
+
+// Ensure we're `no_std` when compiling for WASM.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::dispatch::DispatchResult;
+use frame_system::ensure_signed;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+#[cfg(test)]
+mod tests;
+
+/// A type alias for the balance type from this pallet's point of view.
+type BalanceOf<T> = <T as pallet_balances::Config>::Balance;
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: pallet_balances::Config + frame_system::Config {
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
```

Comment:
Let's have a few `Get` items here? and then provide a default for them?

Reply:
So that we can both have an example of how to provide a default, and how to use one (eg what you are doing for system)